### ISSUE TITLE
API proposal: Utf8.IsValid(bytes)

### ISSUE THREAD

**Issue by GrabYourPitchforks** [dotnet/runtime#502]
See https://github.com/dotnet/corefxlab/issues/2792#issuecomment-561357620 for some further discussion. Many of our UTF-8 utility methods are internal for use only by `Utf8String` or `UTF8Encoding`, but per that discussion there are some scenarios where callers might find those utility methods helpful.

In particular, we should expose an API which answers "given a `ROS<byte>`, please return a `bool` telling me if the input was well-formed UTF-8." It could live on the existing `System.Text.Unicode.Utf8` utility type.

```cs
namespace System.Text.Unicode
{
    // existing static type that shipped in Core 3.0
    public static class Utf8
    {
        // proposed new API
        public static bool IsValid(ReadOnlySpan<byte> value);
    }
}
```

**Comment by neon-sunset** [comment]
Because the other proposal has been closed as duplicate, please consider changing `IsWellFormed(...)` to `IsValid(...)` which matches `Ascii.isValid(...)`. Thank you.

**Comment by jeffhandley** [comment]
_@neon-sunset, thanks for that suggestion--@GrabYourPitchforks and I were just double-checking the existing API name to make them consistent. We've updated the proposal above to capture that, and we've made the proposal more targeted._

We want to look back at this proposal and determine if it should be included in our next round of planning. We already have an implementation of this that is internal on `Utf8Utility`. Marking this as https://github.com/dotnet/runtime/labels/api-ready-for-review.

https://github.com/dotnet/runtime/blob/a9f3262175253dd69eea72007347cd0cda9e6af0/src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.cs#L25-L35

**Comment by bartonjs** [comment]
Looks good as proposed

```C#
namespace System.Text.Unicode
{
    // existing static type that shipped in Core 3.0
    public static class Utf8
    {
        // proposed new API
        public static bool IsValid(ReadOnlySpan<byte> value);
    }
}
```
